django_admin_geomap/__init__.py

```diff
@@ -89,11 +89,20 @@ def set_common(self, request, context):
         return context
 
     def changelist_view(self, request, extra_context=None):
-        """Add geomap data for show at the map."""
+        """Add geomap data to show at the map"""
+
+        # Obtain original response from Django
+        response = super().changelist_view(request, extra_context=extra_context)
+
+        # Obtain final queryset from ChangeList object
+        change_list_queryset = response.context_data['cl'].queryset
+
+        # Add the geomap data to the context
         extra_context = self.set_common(request, extra_context)
-        extra_context[Key.MapItems] = self.get_queryset(request)
+        extra_context[Key.MapItems] = change_list_queryset
+        response.context_data |= extra_context      # add to existing context
```

Comment:
Operator |= does not work in Python 3.7.

I want to keep backwards compatible for this version because it is the last version of Python that can be installed on the Windows7 platform. Many users, especially in the corporate sector, are still using this platform.

So it's better to use dict.update